api/errors.go

```diff
@@ -64,16 +66,29 @@ func HttpCodeForError(err error) int {
 	}
 }
 
-// A simple error handler that renders any error as human-readable JSON to
+// A simple error handler that logs and renders any error as human-readable JSON to
 // the HTTP response stream `w`.
-func HumanReadableJsonErrorHandler(w http.ResponseWriter, r *http.Request, err error) {
-	w.Header().Set("content-type", "application/json; charset=utf-8")
-	w.Header().Set("x-content-type-options", "nosniff")
-	w.WriteHeader(HttpCodeForError(err))
+func HumanReadableJsonErrorHandler(logger log.Logger) func(http.ResponseWriter, *http.Request, error) {
+	return func(w http.ResponseWriter, r *http.Request, err error) {
+		logger.Debug("request failed, handling human readable error",
+			"err", err,
+			"request_id", r.Context().Value(common.RequestIDContextKey),
+			"ctx_err", r.Context().Err(),
+		)
 
-	// Wrap the error into a trivial JSON object as specified in the OpenAPI spec.
-	msg := err.Error()
-	errStruct := apiTypes.HumanReadableError{Msg: msg}
+		// If request context is closed, don't bother writing a response.
+		if r.Context().Err() != nil {
```

Comment:
does this mean we leave the request hanging if our handling times out?

Reply:
If the context is canceled the client already received an EOF (due to the WriteTimeout being reached), so doing anything here will have no effect.


But maybe instead of using our own 'ContextTimeoutHandler' which only cancels the context, we should use upstream [TimeoutHandler](https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/net/http/server.go;l=3620). That one also handles context cancellation, but it also returns a 503, instead of just closing the connection as it is done now. Thats probably nicer.

Reply:
Now that we use `http.TimeoutHandler`, the client receives a 503 and the connection is closed when the timeout is reached. So like before, if the context is canceled there is nothing to do. Even if we didn't return here, the Writes below would fail with `ErrHandlerTimeout`.